### PR TITLE
Add Sentry to `asset-worker` & `router-worker`

### PR DESCRIPTION
**Pull request by penalosa**
## What this PR solves / how to test

Fixes https://jira.cfdata.org/browse/WC-2558 & https://jira.cfdata.org/browse/WC-2684

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because: These workers are not yet tested
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required / Maybe required
  - [x] Not required because: Doesn't affect Wrangler
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because: deployments not yet set up
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Not necessary because: internal detail

### CODE REVIEW

File: packages/workers-shared/asset-worker/src/index.ts [hunk 1]
```diff
@@ -30,15 +36,30 @@ export type ExistsMethod = (pathname: string) => Promise<string | null>;
...
+			);
 			return handleRequest(
 				request,
 				applyConfigurationDefaults(this.env.CONFIG),
 				this.exists.bind(this),
 				this.getByETag.bind(this)
 			);
 		} catch (err) {
-			return new InternalServerErrorResponse(err as Error);
+			const response = new InternalServerErrorResponse(err as Error as Error);
```
Comment (GregBrimble) [inline packages/workers-shared/asset-worker/src/index.ts:55]:
```suggestion
			const response = new InternalServerErrorResponse(err as Error);
```

**Review by GregBrimble — APPROVED** [review]

File: packages/workers-shared/asset-worker/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (CarmenPopoviciu) [inline packages/workers-shared/asset-worker/tsconfig.json:1]: thank you for adding dedicated tsconfig files for each Worker <3

**Review by CarmenPopoviciu — APPROVED** [review]
thank you @penalosa for helping us out with this <3<3<3<3

File: packages/workers-shared/router-worker/src/index.ts [hunk 3]
```diff
@@ -7,18 +8,40 @@ interface Env {
...
+	async fetch(request: Request, env: Env, ctx: ExecutionContext) {
+		let sentry: ReturnType<typeof setupSentry> | undefined;
+
+		try {
+			sentry = setupSentry(
+				request,
+				ctx,
+				env.SENTRY_DSN,
+				env.SENTRY_ACCESS_CLIENT_ID,
+				env.SENTRY_ACCESS_CLIENT_ID
```
Comment (jbergstroem) [inline packages/workers-shared/router-worker/src/index.ts:28]: Perhaps:
```suggestion
				env.SENTRY_ACCESS_CLIENT_SECRET
```

File: packages/workers-shared/asset-worker/src/index.ts [hunk 4]
```diff
@@ -1,56 +1,119 @@
...
+	async fetch(...[request]: Parameters<FetchMethod>): ReturnType<FetchMethod> {
+		let sentry: ReturnType<typeof setupSentry> | undefined;
 		try {
-			return this.handleRequest(request);
+			sentry = setupSentry(
+				request,
+				this.ctx,
+				this.env.SENTRY_DSN,
+				this.env.SENTRY_ACCESS_CLIENT_ID,
+				this.env.SENTRY_ACCESS_CLIENT_ID
```
Comment (jbergstroem) [inline packages/workers-shared/asset-worker/src/index.ts:46]: Here too
```suggestion
				this.env.SENTRY_ACCESS_SECRET
```

**Pull request merged**